galley/pkg/config/analysis/analyzers/gateway/gateway.go

```diff
@@ -111,6 +111,7 @@ func (*IngressGatewayPortAnalyzer) analyzeGateway(r *resource.Entry, c analysis.
 		// the Istio system ingress gateway complain about a missing referenced resource.  (We
 		// don't want to complain about missing system resources, because a user may want to analyze
 		// only his own application files.)
+		// https://github.com/istio/istio/issues/19579 should make this unnecessary
 		if len(gw.Selector) != 1 || gw.Selector["istio"] != "ingressgateway" {
```

Comment:
didn't notice this until now. This should be egress or ingress right?

The label here is entirely arbitrary (but default) so you may false negatives and positives. The proper way to determine is from the args:

```
  containers:
  - args:
    - proxy
    - router
```

router == gateway, sidecar == normal pod.

Reply:
Actually I realized I read this wrong.. thats how you can tell a pod is a gateway though

Reply:
We probably do want to support both egress and ingress. For now I'm mainly trying to match the logic in [our other gateway analyzer](https://github.com/istio/istio/blob/master/galley/pkg/config/analysis/analyzers/gateway/gateway.go#L114-L117), and then the referenced issue tracks a fix that should apply to both.